coroutines-bluebird.js

```diff
@@ -29,8 +29,14 @@ const walk = bluebird.coroutine(function*(dir) {
 
 const main = () => {
   walk(process.argv[2])
-    .catch(s => process.stderr.write(s))
-    .then(s => process.stdout.write(s));
+  .then(data => {
+          process.stdout.write(data);
+          process.exit(0);
+        },
+        err => {
+          process.stderr.write(String(err) + '\n');
+          process.exit(1);
+        });
```

Comment:
The success handler should not throw in this case.

Providing two functions to `.then` highlights a similarity between Promise and Task.